<launch>
  <!-- Wrist diff controller -->
  <group ns="test_controller" clear_params="true">
    <param name="type" type="string" value="WristDifferenceController" />
    <param name="timeout" type="double" value="30.0" />
    <param name="roll_velocity" type="double" value="1.5" />
    <param name="flex_position" type="double" value="1.0" />
    <param name="tolerance" type="double" value="0.20" />
    <param name="sd_max" type="double" value="0.2" />
    <param name="right_effort" type="double" value="-0.54" />
    <param name="left_effort" type="double" value="0.54" />
    <param name="flex_tolerance" type="double" value="0.1" />
    <param name="flex_max" type="double" value="0.1" />

    <group ns="flex_position_controller" clear_params="true">
      <param name="type" type="string" value="JointPositionController" />
      <param name="joint" type="string" value="r_wrist_flex_joint" />
      <param name="pid/p" type="double" value="10" />
      <param name="pid/i" type="double" value="10" />
      <param name="pid/d" type="double" value="0.0" />
      <param name="pid/i_clamp" type="double" value="0.3" />
    </group>
    
    <group ns="roll_velocity_controller" clear_params="true">
      <param name="type" type="string" value="JointVelocityController" />
      <param name="joint" type="string" value="r_wrist_roll_joint" />
      <param name="pid/p" type="double" value="2.0" />
      <param name="pid/i" type="double" value="10.0" />
      <param name="pid/d" type="double" value="0.0" />
      <param name="pid/i_clamp" type="double" value="0.1" />
    </group>
  </group>

  <node pkg="pr2_controller_manager" type="spawner.py"
        args="test_controller" name="wrist_diff_ctrl" />

  <!-- Analysis -->
  <node pkg="qualification" type="wrist_diff_analysis.py"
        name="wrist_diff" output="screen" />
</launch>
